k8s/upgrades/0.9.0-1.0.0/cstor_pool_upgrade.sh

```diff
@@ -0,0 +1,279 @@
+#!/bin/bash
+
+###########################################################################
+# STEP: Get SPC name and namespace where OpenEBS is deployed as arguments #
+#                                                                         #
+# NOTES: Obtain the pool deployments to perform upgrade operation         #
+###########################################################################
+
+pool_upgrade_version="v1.0.x-ci"
+current_version="0.9.0"
+
+function error_msg() {
+    echo "Failed to upgrade pool $spc. Please make sure that the pool $spc upgrade should be successful before moving to the next step"
+}
+
+function usage() {
+    echo
+    echo "Usage:"
+    echo
+    echo "$0 <spc-name> <openebs-namespace>"
+    echo
+    echo "  <spc-name> Get the SPC name using: kubectl get spc"
+    echo "  <openebs-namespace> Get the namespace where pool pods"
+    echo "    corresponding to SPC are deployed"
+    exit 1
+}
+
+##Checking the version of OpenEBS ####
+function verify_openebs_version() {
+    local resource=$1
+    local name_res=$2
+    local openebs_version=$(kubectl get $resource $name_res -n $ns \
+                 -o jsonpath="{.metadata.labels.openebs\.io/version}")
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Failed to get version from $resource: $name_res | Exit Code: $rc"
+        error_msg
+        exit 1
+    fi
+
+    if [[ $openebs_version != $current_version ]] && \
+          [[ $openebs_version != $pool_upgrade_version ]]; then
+        echo "Expected version of $name_res in $resource is $current_version but got $openebs_version";exit 1;
+    fi
+    echo $openebs_version
+}
+
+## get_csp_list will return the csp list related corresponding spc
+function get_csp_list() {
+    local csp_list=""
+    local spc_name=$1
+    csp_list=$(kubectl get csp \
+               -l openebs.io/storage-pool-claim=$spc_name \
+               -o jsonpath="{range .items[*]}{@.metadata.name}:{end}")
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Failed to get csp related to spc $spc"
+        error_msg
+        exit 1
+    fi
+    echo $csp_list
+}
+
+function make_spc_blockdevice_list() {
+    local spc_bd_list=$1
+    local disk_name=$2
+    ## For sparse block device name looks like sparse-1234567
+    local bd_name=$(echo $disk_name | sed 's|disk-|blockdevice-|g')
+
+    if [ -z $spc_bd_list ]; then
+        echo "\"$bd_name\""
+    else
+        echo "$spc_bd_list,\"$bd_name\""
+    fi
+}
+
+## patch_blockdevice_list_for_spc to patch block device changes related to spc
+function patch_blockdevice_list_for_spc() {
+    local spc_name=$1
+
+    local spc_disk_list=$(kubectl get spc $spc_name \
+                          -o jsonpath='{.spec.disks.diskList}' | \
+                          tr "[]" " ")
+    local spc_bd_list=""
+    local no_of_blockdevices=$(echo $spc_disk_list | wc -w)
+
+    ##Patch SPC only if it is manual provisioning
+    if [ $no_of_blockdevices != 0 ]; then
```

Comment:
'else' should be error?

Reply:
In auto provisioning case we won't have any disks